webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/disassociate/BulkDisassociateModal.js

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Modal,
+  Alert,
+  Button,
+  TextContent,
+  Text,
+  TreeView,
+} from '@patternfly/react-core';
+import { addToast } from '../../../ToastsList/slice';
+import { foremanUrl } from '../../../../common/helpers';
+import { translate as __ } from '../../../../common/I18n';
+import { BULK_DISASSOCIATE_KEY, bulkDisassociate } from './actions';
+import { APIActions } from '../../../../redux/API';
+import {
+  HOSTS_API_PATH,
+  API_REQUEST_KEY,
+} from '../../../../routes/Hosts/constants';
+import { failedHostsToastParams } from '../helpers';
+
+const BulkDisassociateModal = ({
+  isOpen,
+  closeModal,
+  selectAllHostsMode,
+  selectedCount,
+  selectedResults,
+  fetchBulkParams,
+}) => {
+  const dispatch = useDispatch();
+  const virtualHosts = selectedResults?.filter(
```

Comment:
Perhaps also avoid virtual vs physical in variable naming?
```suggestion
  const hostsWithComputeResource = selectedResults?.filter(
```